lib/ex_doc/nodes.ex

```diff
@@ -106,6 +109,7 @@ defmodule ExDoc.TypeNode do
           signature: nil | String.t(),
           source_url: nil | String.t(),
           source_path: nil | String.t(),
-          annotations: list()
+          annotations: list(),
+          module: nil | String.t()
```

Comment:
Do we really need this new field? We should always know the module it belongs to based on the context. For example, for summary_entry_template, we can probably pass it when invoking it here: https://github.com/elixir-lang/ex_doc/blob/59905ee0415cf639136001988f502bd71872dd7a/lib/ex_doc/formatter/html/templates/summary_template.eex#L6

Reply:
I had thought that each node may use some info about its containing module, however, passing the module through makes sense. I take out the field and change the template signatures. Thanks!